### PR TITLE
feat(pie-webc): DSW-1962 add script to automate exporting components

### PR DESCRIPTION
**Pull request by jamieomaguire**
## Describe your changes (can list changeset entries if preferable)
---
"@justeattakeaway/pie-webc": minor
---

[Added] - An automation script named `add-components` which will add all pie components into pie-webc
[Added] - Supporting unit tests

---
"@justeattakeaway/generator-pie-component": minor
---

[Changed] - Invoke the new `add-components` command after generating new components to ensure they are correctly added to pie-webc
[Changed] - Improve logging and message formatting

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have used the `/snapit` functionality to test my changes in a consuming application

### CODE REVIEW

**Review by siggerzz — APPROVED** [review]

**Review by siggerzz — DISMISSED** [review]

File: packages/components/pie-webc/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "@justeattakeaway/pie-components-config/tsconfig.json",
```
Comment (xander-marjoram) [inline packages/components/pie-webc/tsconfig.json:2]: Not sure if this is still needed, we can chat when you're back

File: packages/tools/generator-pie-component/src/app/index.ts [hunk 2]
```diff
@@ -48,6 +48,8 @@ export default class extends Generator {
     }
 
     async end () {
-        this.log(chalk`Your component has been created at ${this.props.componentPath}`);
```
Comment (xander-marjoram) [inline packages/tools/generator-pie-component/src/app/index.ts:51]: Not sure if it was just a Windows thing, but for me this was logging as:
```
Your component has been created at , packages/components/pie-test/
```
which looked a bit messy...

File: packages/tools/generator-pie-component/src/app/index.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ export default class extends Generator {
     props: Props;
 
     async initializing () {
-        this.log(chalk`A Yeoman generator for Pie Web Components`);
+        this.log(chalk.hex('#f36805')('Starting PIE web component generator...'));
```
Comment (xander-marjoram) [inline packages/tools/generator-pie-component/src/app/index.ts:12]: Brand orange! 🧡

**Review by dandel10n — DISMISSED** [review]
Great work! 🔥

File: packages/components/pie-webc/src/componentService.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        const jsFilePath = this.path.join(target.dir, `${componentName}.js`);
+        const tsFilePath = this.path.join(target.dir, `${componentName}.d.ts`);
+
+        const fileContent = `export * from '${target.exportPath}';\n`;
+        this.fs.writeFileSync(jsFilePath, fileContent);
+        this.fs.writeFileSync(tsFilePath, fileContent);
+    }
+
+    writePackageJson (path, content) {
+        this.fs.writeFileSync(path, `${JSON.stringify(content, null, 2)}\n`);
```
Comment (fernandofranca) [inline packages/components/pie-webc/src/componentService.js:80]: Glad that the extra line was added at the end! 👍

File: packages/components/pie-webc/src/componentService.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
+import chalk from 'chalk';
+
+export class ComponentService {
```
Comment (fernandofranca) [inline packages/components/pie-webc/src/componentService.js:3]: More of a general question, how do we feel about using (or not) TypeScript for utility/script code, like this class?

**Review by fernandofranca — DISMISSED** [review]
🔥 🔥 🔥 Good automation stuff!!!

File: packages/components/pie-webc/src/index.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env node
+import fs from 'fs';
+import path from 'path';
+import chalk from 'chalk';
+
+import { ComponentService } from './componentService.js';
+
+const main = (fs, path) => {
```
[the remark this answers is not in the document]
Comment (xander-marjoram) [reply packages/components/pie-webc/src/index.js:8]: JSDocs is a great shout, I was planning to do this but forgot so thanks for the reminder!

File: packages/components/pie-webc/src/componentService.js (hunk 5 above)
Comment (xander-marjoram) [reply packages/components/pie-webc/src/componentService.js:3]: I'm not agains the idea in principle. I think typings for the node `path` and `fs` modules could be tricky (or very easy) to set up. Having typescript would also mean we have to run the typescript compiler before `npx` can run, I believe.

**Review by fernandofranca — APPROVED** [review]

**Review by dandel10n — APPROVED** [review]

**Pull request merged**